internal/cloudapi/server.go

```diff
@@ -0,0 +1,222 @@
+		if err != nil {
+			http.Error(w, fmt.Sprintf("Failed to depsolve build packages for %s/%s/%s: %s", ir.ImageType, ir.Architecture, request.Distribution, err), http.StatusBadRequest)
+			return
+		}
+
+		imageOptions := distro.ImageOptions{Size: imageType.Size(0)}
+		if request.Customizations != nil && request.Customizations.Subscription != nil {
+			imageOptions.Subscription = &distro.SubscriptionImageOptions{
+				Organization:  request.Customizations.Subscription.Organization,
+				ActivationKey: request.Customizations.Subscription.ActivationKey,
+				ServerUrl:     request.Customizations.Subscription.ServerUrl,
+				BaseUrl:       request.Customizations.Subscription.BaseUrl,
+				Insights:      request.Customizations.Subscription.Insights,
+			}
+		}
+
+		manifest, err := imageType.Manifest(nil, imageOptions, repositories, packages, buildPackages)
+		if err != nil {
+			http.Error(w, fmt.Sprintf("Failed to get manifest for for %s/%s/%s: %s", ir.ImageType, ir.Architecture, request.Distribution, err), http.StatusBadRequest)
+			return
+		}
+
+		imageRequests[i].manifest = manifest
+
+		if len(*ir.UploadRequests) != 1 {
+			http.Error(w, "Only compose requests with a single upload target are currently supported", http.StatusBadRequest)
+			return
+		}
+		uploadRequest := (*ir.UploadRequests)[0]
+		/* oneOf is not supported by the openapi generator so marshal and unmarshal the uploadrequest based on the type */
+		if uploadRequest.Type == "aws" {
+			var awsUploadOptions AWSUploadRequestOptions
+			jsonUploadOptions, err := json.Marshal(uploadRequest.Options)
+			if err != nil {
+				http.Error(w, "Unable to marshal aws upload request", http.StatusInternalServerError)
+				return
+			}
+			err = json.Unmarshal(jsonUploadOptions, &awsUploadOptions)
+			if err != nil {
+				http.Error(w, "Unable to unmarshal aws upload request", http.StatusInternalServerError)
+				return
+			}
+
+			key := fmt.Sprintf("composer-cloudapi-%s", uuid.New().String())
+			t := target.NewAWSTarget(&target.AWSTargetOptions{
+				Filename:        imageType.Filename(),
+				Region:          awsUploadOptions.Region,
+				AccessKeyID:     awsUploadOptions.S3.AccessKeyId,
+				SecretAccessKey: awsUploadOptions.S3.SecretAccessKey,
+				Bucket:          awsUploadOptions.S3.Bucket,
+				Key:             key,
+			})
+			if awsUploadOptions.Ec2.SnapshotName != nil {
+				t.ImageName = *awsUploadOptions.Ec2.SnapshotName
+			} else {
+				t.ImageName = key
+			}
+
+			targets = append(targets, t)
+		}
+	}
+
+	var ir imageRequest
+	if len(imageRequests) == 1 {
+		// NOTE: the store currently does not support multi-image composes
+		ir = imageRequests[0]
+	} else {
+		http.Error(w, "Only single-image composes are currently supported", http.StatusBadRequest)
+		return
+	}
+
+	id, err := server.workers.Enqueue(ir.manifest, targets)
+	if err != nil {
+		// This is a programming errror.
```

Comment:
This is not trrrrrue. When using `fsjobqueue`, this returns write errors.

Reply:
Is it still panic-worthy though?

Reply:
I think an internal server error would be cleaner.